node.js/index.js

```diff
@@ -0,0 +1,114 @@
+// External Libraries:
+const agent = require('agentkeepalive');
+const request = require('request');
+
+// Internal Libraries:
+const constants = require('./constants');
+
+// Getting Environment Variables:
+const getConfig = () => {
+    var config = {};
+
+    if (process.env.LOGDNA_KEY) config.key = process.env.LOGDNA_KEY;
+    if (process.env.LOGDNA_HOSTNAME) config.hostname = process.env.LOGDNA_HOSTNAME;
+
+    if (process.env.LOGDNA_TAGS && process.env.LOGDNA_TAGS.length > 0) {
+        config.tags = process.env.LOGDNA_TAGS.split(',').map(tag => tag.trim()).join(',');
+    }
+
+    return config;
+};
+
+// Message Sanity Check:
+const sanitizeMessage = message => {
+    var sanitized = message.substring(0, constants.MAX_LINE_LENGTH);
+    if (sanitized !== message) sanitized += ' (cut off, too long...)';
+    return sanitized;
+};
+
+// Parsing the gzipped Log Data:
+const parseEvent = event => {
+    return JSON.parse(require('zlib').unzipSync(Buffer.from(event.awslogs.data, 'base64')));
+};
+
+// Preparing the Messages and Options:
+const prepareLogs = eventData => {
+    return eventData.logEvents.map(event => {
+        return {
+            line: JSON.stringify({
+                message: sanitizeMessage(event.message)
+                , source: 'CloudWatchLogs'
+                , event: {
+                    type: eventData.messageType
+                    , id: event.id
+                }, log: {
+                    group: eventData.logGroup
+                    , stream: eventData.logStream
+                }
+            }), timestamp: event.timestamp
+            , file: eventData.logStream
+            , meta: {
+                owner: eventData.owner
+                , filters: eventData.subscriptionFilters
+            }
+        };
+    });
+};
+
+// Shipping the Logs:
```

Comment:
As mentioned elsewhere, your verb tense is inconsistent throughout your comments. I personally prefer imperative ("Ship the logs"). You could choose 3rd person if you like ("Ships the logs") but i would avoid "-ing", it reads strange to me.

Reply:
This is debatable - as long as there is consistency, it should be fine, I think - and it's also straightforward for the developer who is reading and trying to understand.